webrtc/src/data_channel/mod.rs

```diff
@@ -67,10 +72,10 @@ pub struct RTCDataChannel {
     // is created, the binaryType attribute MUST be initialized to the string
     // "blob". This attribute controls how binary data is exposed to scripts.
     // binaryType                 string
-    pub(crate) on_message_handler: Arc<Mutex<Option<OnMessageHdlrFn>>>,
-    pub(crate) on_open_handler: Arc<Mutex<Option<OnOpenHdlrFn>>>,
-    pub(crate) on_close_handler: Arc<Mutex<Option<OnCloseHdlrFn>>>,
-    pub(crate) on_error_handler: Arc<Mutex<Option<OnErrorHdlrFn>>>,
+    pub(crate) on_message_handler: Arc<ArcSwapOption<Mutex<OnMessageHdlrFn>>>,
```

Comment:
why do we need Arc inside another Arc here?

Reply:
`ArcSwapOption` is basically an `Option<Arc<T>>` not `Arc<Option<Arc<T>>>`, it has no `Clone` implementation, and must be wrapped into another `Arc` if it needs to be shared. We can only clone the inner `Arc`, but it won't be changed if we call `ArcSwapOption::store` after.